src/python/pants/reporting/reporting.py

```diff
@@ -88,9 +95,24 @@ def initialize(self, run_tracker, all_options, start_time=None):
 
     # Set up Zipkin reporting.
     zipkin_endpoint = self.get_options().zipkin_endpoint
+    trace_id = self.get_options().trace_id
+    parent_id = self.get_options().parent_id
+    if all((trace_id, parent_id)) and zipkin_endpoint is None:
```

Comment:
This would probably be a little clearer as:
```python
if zipkin_endpoint is None:
  if trace_id is not None or parent_id is not None:
    # error
else:
  if trace_id is None ^ parent_id is None:
    # error
```

Reply:
We do not want to raise an error when all three flags a None. With the first if it could happen

Reply:
```
if zipkin_endpoint is not None:
      if trace_id is None and parent_id is not None or parent_id is None and trace_id is not None:
        raise ValueError("Flags trace_id and parent_id must both either be set or not set.")
    else:
      if trace_id is not None and parent_id is not None:
        raise ValueError("The zipkin_endpoint flag must be set if trace_id and parent_id are given.")
      if trace_id is None and parent_id is not None:
        raise ValueError("Falgs trace_id and zipkin_endpoint must be set if parent_id is given.")
      if parent_id is None and trace_id is not None:
        raise ValueError("Flags parent_id and zipkin_endpoint must be set if trace_id is given.")
```